### ISSUE TITLE
Thursday

### ISSUE THREAD

**Issue by Codingmaestre2015** [WestMEC-Coding/Week-11-Bell-Work#3]

**Comment by Codingmaestre2015** [comment]
Please describe the Document Object Model, in detail. If you would like, you may draw a diagram, snap a photo with your phone, and upload it to Git.

Please, do NOT just Google Image search the DOM hierarchy and copy and paste it.

You will also have to use text to describe the DOM... not just a diagram.

**Comment by clocksin-nick-wm** [comment]
DOM is a way of describing in code how to look for a specific piece of information. Let's say I wanted to run a set of elements in a code. First I would link the starting file to the file that contains the element, and then connect and other subfolders/files that contain the elements to find the individual element to display in the code. Another good example is on windows when you are looking for a file. When you go into each sub folder to find that file, you are linking the previous files together so that the sources location is known.

**Comment by steele-ariel-wm** [comment]
I don't quite understand DOM yet.

**Comment by pierce-eric-wm** [comment]
Dom is somewhat like a file path were you have it going from the drive to the folder to the item that you want. So on my computer at home if i wanted to go to a game folder i would have to go to E/game/gamefolder think of the / as a period and that is how a dom is set up.

**Comment by mason-gavin-wm** [comment]
The Document Object Model is first a language-independent convention for representing and interacting with objects in documents. It allows a user to directly go to a certain document in their files and manipulate them in any way they want, basically like a file path to something you want.

**Comment by burton-jacob-wm** [comment]
DOM's are like a locating system. First it looks for the parent, then its child, and then its child. It goes in order from left to right to locate certain parts of the website or document. 

Document > getElementbyID#ID1 > textElement > style

**Comment by capp-brendan-wm** [comment]
The DOM is a programming interface that defines the structure and how a document is used and accessed, it can represent many forms of different types of information that can be stored in diverse systems, it basically manages all the data put into it.

**Comment by breeah-wm** [comment]
Document Object Model is used to find and interact with document models like HTML and XML.

**Comment by mcclure-calvin-wm** [comment]
DOM is a method for communicating with html to find a certain section of code. This can be used in tandem with JavaScript to change certain pieces of html when a certain condition is met. This is done in a similar way as files would be navigated

Ex 
session1/desktop/js_basics_hw/h1.html    ~=     document.text.style.color

**Comment by Keyan-Ngai-wm** [comment]
The document object model is an application programming interface for valid HTML. It defines the structure of documents and the way a document is accessed and used. With the Document Object Model, programmers can build documents, navigate their structure, and add, change, or delete content. Anything found in an HTML document can be accessed, changed, deleted, or added using the Document Object Model.

**Comment by schweikert-zachary-wm** [comment]
window.document.body.p
                                      .table.tr.td
                              head.
I wrote this as a flow chart. You can choose either body or head, and then p or table if you choose body. You choose one depending on which one you need to change or call.

**Comment by brey-matthew-wm** [comment]
A document object model is a map of a webpages html that can be accessed at any time with javascript. There are several methods of calling a part of the model with javascript, but the most common is document.GetElementById("The Id").innerHTML = "Hello World";. The get element command finds the element described, and the .innerHTML command can change, or add something to the inner HTML that you called with the GetElement command.

**Comment by thorpe-sean-wm** [comment]
DOM is used to narrow down to a certain section of code (like a file path) and allows programs or scripts to access and update whats in it, its structure, or its style

for example in code it would look like this:
Document> html> body> h1> "My Company"

**Comment by cunningham-corbin-wm** [comment]
The DOM is a way to change your HTML and CSS properties after the webpage has been loaded. It is kind of like finder in the fact that you access the "folders" through selecting them. It is kind of like a file path except your files are  instead elements in HTML and CSS.

Example:
Document.getElementByID("UserName").innerHTML = "Hello," + " " + currentUser;

**Comment by riggs-jacob-wm** [comment]
The Document Object Model is a hierarchical structure of nodes, where there is one for each tag and block of text. In simpler terms, it is a file path where you can manipulate things inside.

**Comment by Garcia-Rogelio-WM** [comment]
DOM was a way to connect JS with HTML or to link it in a path. an example could be finder.session1.PHPstorm projects.Project

**Comment by shephard-azaniyah-wm** [comment]
A DOM is the place of which titles of objects are presented while the deeper text behind each box is the action taken and the effect it does. It is usually seen as a tree chart indicating the top power and dividing it out into smaller parts. This also helps organize and find any issues.

**Comment by hogencamp-devon-wm** [comment]
DOM is the standard object model used to
-Set HTML elements as objects
-Change properties of HTML elements
-Defines methods to access elements
-Defines the events for elements

var myElement = document.getElementById("intro");

This is an example of DOM, we can use variables in JS to set and change what appears on the page. The get element by id is just how you find the element. You give an element an id so JS can find it on the page.

**Comment by osborn-daucen-wm** [comment]
DOM is the way a computer locates certain elements of a file. Most of the time it goes from a window to a document to a form and then to a certain element that you need to display, change, etc. The document object model is necessary to alter elements of your HTML or CSS through Javascript and more commonly used to make a website that is adaptable to multiple users.

**Comment by casey-leevon-wm** [comment]
Dom makes everything in a webpage into an object and therefore editable. It connects Javascript with HTML.

**Comment by diaz-andy-wm** [comment]
The DOM or Document Object Model is a method of viewing code as just a bunch of objects. These objects which are to be created in HTML can be manipulated through Javascript programming. DOM manipulation is to be done a certain way. Only by calling on an object's whole location, can an object be properly called and files within files must be separated from the parent file with periods.

**Comment by deherrera-marques-wm** [comment]
The Document Object Model is a locating system, anything in the HTML document can be changed in this model, because it connects  Javascript with HTML.

Ex:
Document> html> body> h1> "Microsoft"

**Comment by le-ben-wm** [comment]
The Document Object Model is used to reference HTML code in JavaScript. It is used to dynamically change HTML based on user input, whether it is a small part of the code or the entire code. To use it, you would first reference the largest object, which in most cases is "document", and then based on what you are looking to reference, you would use getElementById, getElementByClass, or another variation of these to call up a certain part of the HTML code. These parts are connected by periods.
ex: 
var x = document.getElementById("names");
console.log(x);

This code would print everything under the id "names" to the console.

**Comment by gonzalez-jose-wm** [comment]
A Document Object Model is the logical structure of documents and the way a document is accessed and manipulated. Don't know to much yet other than that at the moment.

**Comment by marquez-mollee-wm** [comment]
Im not to sure what the DOM is but I thinks it's a way to locate information i an HTML file.
Window.PhpstormProject.data_website

**Comment by athina-bitsui-wm** [comment]
a Document Object Model (DOM)t is a language and a cross-platform for HTML, XHTML, and XML.

**Comment by warren-evan-westmec** [comment]
The DOM lets you manage and edit different parts of a webpage all at once

**Comment by lopez-efrain-wm** [comment]
I don't really understand the dom yet, all i know is that it is used to change things in html through java.

**Comment by turner-dalton-wm** [comment]
It is the api for html and javascript to communicate and access each other. Documents are made of objects and each object has nodes in it.

**Comment by baker-morgan-wm** [comment]
The DOM is a way to break down information on a web page. Going from window.document etc.

**Comment by timothy-david-wm** [comment]
DOM is the standard object model used to
<ul>
<li>Set HTML elements as objects</li>
<li>Change properties of HTML elements</li>
<li>Defines methods to access elements</li>
<li>Defines the events for elements</li>
</ul>
var myElement = document.getElementById("intro");

This is an example of DOM, we can use variables in JS to set and change what appears on the page. The get element by id is just how you find the element. You give an element an id so JS can find it on the page.

**Comment by shumway-kyle-wm** [comment]
The dom is a cross platform reference for attributes , elements and text

**Comment by Dunn-Carrie-wm** [comment]
Dom is the way you manage things or edit things in javascript. I don't really understand it. But from the lesson yesterday i gathered it was the way to edit java script on a certain feature.

**Comment by chopra-austin-west-mec** [comment]
DOM is a way of changing Items on a webpage using JS


image of DOM 
![image](https://cloud.githubusercontent.com/assets/13738866/10529628/8621e712-7354-11e5-9de7-3b211d505b0c.png)

**Comment by weitzel-scotland-wm** [comment]
The DOM (Document Object Model) is a hierarchy of nodes in HTML that JavaScript can use to manipulate different pieces of the webpage. ex.

Window.Document.Element.Text

**Comment by casselman-bailey-wm** [comment]
The Document Object Model is a structure of nodes. It is the way things are managed in html, css, and javascript. Its basically a way to organize common terminology.

**Comment by spiller-taylor-wm** [comment]
The DOM treats every tag on the webpage as a tag in an object. It further stores all attributes as well. This allows for easy access to any part of a webpage that you may want to restyle or change.

**Comment by Tucker-Cole-wm** [comment]
The DOM is basically the blocks that make up a web page.  The blocks are the categories within a page that categorize elements.  It also makes javascript used in html.

**Comment by iniguez-angel-wm** [comment]
Document-The whole structure of a page.
Object- The objects with in that structure.
Model- the blueprint to it all.

**Comment by navarro-aiden-wm** [comment]
The Document Object Model is a way to view code as just objects. These objects, which are created in HTML, can be manipulated through Javascript functions. The DOM manipulation is done a certain way. Only by calling on an object's whole location, can an object be properly called and files within files must be separated from the parent file using periods it separate them.

**Comment by chu-christopher-wm** [comment]
DOM reaches into your webpage and allows change to nodes/elements.
var stuff = document.getElementById("body");

**Comment by moncada-carlos-WM** [comment]
I don't really understand DOM. but it's just nodes

**Comment by kurtz-zachary-wm** [comment]
The DOM also known as the Document Object Model, is a cross platform reference for attributes, through HTML and JavaScript. It can be used to change pieces of a webpage. 
Window -> Document -> Element -> Text

**Comment by adenola-ahmed-wm** [comment]
It is a cross-platform for HTML.It lets you manipulate parts of the webpage

**Comment by murillo-daisy-wm** [comment]
The Document Object Model is a break down of the code; it structures components.

**Comment by chavez-juan-wm** [comment]
The DOM is a standard for accessing documents. It allows programs to dynamically access and update the content, structure, and style of a document.

**Comment by ErickCiudad** [comment]
The document object model is a way of locating elements of html and changes them based on interactions or inputs through javascript. It branches out in a tree like structure with different "nodes" on every branch.

**Comment by yousif-zues-wm** [comment]
The DOM is a universal label for each object within a document (a webpage or the syntax).
It is used to point out certain areas of code to others without having to describe it at all.
There are many types of labels but the most basic and common are known as "Nodes" Nodes hold any sort of data wether it can be text images attributes, any data at all.
![imag0105](https://cloud.githubusercontent.com/assets/13738853/10531154/6b2508a4-7363-11e5-93aa-1ad9cb448690.jpg)
